### ISSUE TITLE
Malformed translation files in Maniac Mansion Mania games (maybe others)

### ISSUE THREAD

**Issue by dreammaster** [adventuregamestudio/ags#1332]
In at least the two games I tested, MMM 2 and MMM 8, the game aborts with an error if you try to use the English translation. In MMM 8, for example, the error says overlapping block, expected to end at offset 46, finished reading at 47. This one byte difference is the same in MMM 2 as well.

Although it's a bit of a hack, I put in a special if check in the following commit:
https://github.com/scummvm/scummvm/commit/bf185107272d7c07da36988dd56d8a6112a52edf

It's not ideal, because it doesn't limit the check to only .tra files, but I figured since I hadn't gotten any block errors in other games I tested at all, explicitly checking for the case of a one byte difference at the very start of the file was sufficiently unique to still allow for errors happening anywhere else.

**Comment by ivan-mogilko** [comment]
@dreammaster it seems that this may be the problem with every translation created before AGS 3.0. I tried few older editors (2.72, 2.7, etc), and they all create a translation file that fails this check in our current master.

My guess is that maybe these editors had a block write miscalculation and saved one more byte at the end of the "gamename" field.

Unfortunately the translation file does not have any "format version", but maybe it's okay to rely on loaded game version and add +1 to the block size when reading translations.

**Comment by ivan-mogilko** [comment]
@dreammaster I did a change that we discussed on discord, reorganizing block reading function(s) into a reader class.
This is of course not only for this bug, but I believe it's generally more flexible and thus easier to customize for workarounds.

These are commits: 462535ec21e2b184651d03d118b9cfd2258520e4, 838605f7689febdf8174083d9dace140a6e07e19 and 92c0844519653507ec979037d46a58a60ea93032
the last one is for malformed pre-3.* translation files.
